src/ui/components/shared/LoadingScreen.tsx

```diff
@@ -31,72 +26,43 @@ export function LoadingScreenTemplate({
   );
 }
 
-// White progress screen used for showing the scanning progress of a replay
-export function ProgressBar({ progress }: { progress: number }) {
-  // We keep a low-water mark here, because it's possible for the amount that we
-  // have loaded to actually go *down* due to new unprocessed regions
-  const [minProgress, setMinProgress] = useState<number>(0);
-
-  const displayedProgress = useMemo(() => {
-    if (!progress) {
-      return minProgress;
-    }
-
-    if (progress > minProgress) {
-      setMinProgress(progress);
-    }
-
-    return Math.max(minProgress, progress);
-  }, [progress, minProgress]);
-
-  return (
-    <div className="relative h-1.5 w-full overflow-hidden rounded-lg bg-themeBase-90 p-0">
-      <div
-        className="t-0 absolute h-full bg-primaryAccent"
-        style={{ width: `${displayedProgress}%`, transitionDuration: "400ms" }}
-      />
-    </div>
-  );
-}
-
-function LoadingScreen({ uploading, awaitingSourcemaps, progress }: PropsFromRedux) {
+function LoadingScreen({
+  uploading,
+  awaitingSourcemaps,
+  fallbackMessage,
+  stalledTimeout = 2000,
+}: PropsFromRedux & { fallbackMessage: string; stalledTimeout?: number }) {
   // The backend send events in this order: uploading replay -> uploading sourcemaps.
+  let waitingForMessage = <span>{fallbackMessage}</span>;
   if (awaitingSourcemaps) {
-    return (
-      <LoadingScreenTemplate>
-        <span>Uploading sourcemaps</span>
-      </LoadingScreenTemplate>
-    );
+    waitingForMessage = <span>Uploading sourcemaps...</span>;
```

Comment:
I really don't like that these two conditions (uploading recording data and uploading source maps) live here instead of the upload screen. That feels like the place they *should* be, but this PR was big already just trying to get the E2E tests to work, so I don't want to add on anymore stuff.